### ISSUE TITLE
Hydration completed but contains mismatches

### ISSUE THREAD

**Issue by mroddev** [harlan-zw/nuxt-simple-sitemap#4]
### Describe the bug

Getting a "Hydration completed but contains mismatches" console error when doing a refresh in production.
When the website is open for the first time and even after doing page navigation the warning does not appear. But once I do a refresh then the console error happens.

I do have dynamic routes and they all seem to be generated on build, don't know if that could be the cause.

This problem started right after installing and configuring the module, removing it solves the issue.

### System Info

```Shell
System:
    OS: Windows 10 10.0.19044
  Binaries:
    Node: 16.14.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.5.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chromium (108.0.1462.76)
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/harlan-zw/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by harlan-zw** [comment]
Hey @mroddev 

I can see why you might think this module is responsible for this error, but this module does not generate any markup besides the sitemap.xml file, so I don't see any way in how it could contribute to this error.

You should use this module https://html-validator.nuxtjs.org/, you likely have invalid HTML somewhere causing the hydration issue.

Otherwise, please provide a reproduction I'd be happy to investigate further. Will close for now.

**Comment by mroddev** [comment]
Hello @harlan-zw 

Thanks for your reply. I indeed installed the module you requested and found some issues, but even after fixing them the problem would persist.

I spent some hours debugging my code and finally found the problem, which was totally on my side.

What I found weird is that the hydration error only shows up with the nuxt-simple-sitemap module installed when it should've shown since the beggining.

Overall it's all fixed now, sorry for creating this thread since my problem was not even relevant to this specific module.
